### PR TITLE
Migrate Pact tests to GitHub Actions

### PR DESCRIPTION
**Pull request by ddzz**
## Description of change
This PR migrates the `pact:verify` rake task to GitHub Actions. Pact will be removed entirely from CircleCI in a separate PR.

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/22374

### CODE REVIEW

**Comment by billfienberg** [comment]
Here's a link to the CircleCI config that contains the current pact workflow/jobs/steps/etc:

https://github.com/department-of-veterans-affairs/vets-api/blob/master/.circleci/config.yml

File: .github/workflows/pact.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+name: Pact Contract Tests
+on: [workflow_dispatch]
+jobs:
+  tests:
+    name: Pact Contract Tests
+    runs-on: ubuntu-16.04
+    env:
+      BUNDLE_ENTERPRISE__CONTRIBSYS__COM: ${{ secrets.BUNDLE_ENTERPRISE__CONTRIBSYS__COM }}
+      CI: true
```
[the remark this answers is not in the document]
Comment (ddzz) [reply .github/workflows/pact.yml:9]: Good point, I'll clean this up.

File: .github/workflows/pact.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+name: Pact Contract Tests
+on: [push]
+jobs:
+  tests:
```
Comment (LindseySaari) [inline .github/workflows/pact.yml:4]: Can the name be more specific?

Comment (ddzz) [reply .github/workflows/pact.yml:4]: Fixed

File: .github/workflows/pact.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      - name: Install dependencies (pdftk, poppler, imagemagick)
+        run: sudo apt install imagemagick pdftk poppler-utils
+
+      - name: Setup Database
+        run: bundle exec rake db:schema:load
+
+      - name: Verify stable Pacts
+        run: bundle exec rake pact:verify
+        env:
+          PACT_BROKER_BASIC_AUTH_USERNAME: pact-broker
```
Comment (LindseySaari) [inline .github/workflows/pact.yml:59]: Does the PACT_BROKER_BASIC_AUTH_PASSWORD need to be set here too? Or maybe line 47 takes care of that

Comment (ddzz) [reply .github/workflows/pact.yml:59]: Yeah, the `Get bot token from Parameter Store` step handles that, so it's not needed here.

**Review by LindseySaari — APPROVED** [review]

**Pull request merged**